mutation/frozen_mutation.hh

```diff
@@ -230,8 +232,9 @@ public:
     printer pretty_printer(schema_ptr) const;
 };
 
-frozen_mutation freeze(const mutation& m);
-std::vector<frozen_mutation> freeze(const std::vector<mutation>&);
+// is_preemptible can be set when called in a seastar thread
+frozen_mutation freeze(const mutation& m, is_preemptible = is_preemptible(seastar::thread::running_in_thread()));
+std::vector<frozen_mutation> freeze(const std::vector<mutation>&, is_preemptible = is_preemptible(seastar::thread::running_in_thread()));
```

Comment:
This isn't reliable. We could be in a coroutine or continuation launched from a thread. The caller would not expect us to be interrupted.

```c++
void in_thread() {
    not_in_thread().get();
}

future<> not_in_thread() {
    // This won't yield, right?
    auto ff = freeze(m);
}

```

Reply:
Hmm, `seastar::thread::running_in_thread()` would return true before `not_in_thread` yields, when it's called from a thread, but as far as I checked it does return false when we're in a continuation.  That said, I agree it would be surprising for the caller - so we can leave that with no default assignment (or default to is_preemptible::no for backward compatibility) and let the caller pass is_preemptible::yes when it's sure in a thread.